src/controllers/device.controller.ts

```diff
@@ -0,0 +1,21 @@
+import type {
+  Handler,
+  Request,
+  Response,
+} from 'express';
+import { deviceService } from '../services';
+
+const addDevice: Handler = (
+  req: Request,
+  res: Response,
+): void => {
+  deviceService.addDevice(req.body)
```

Comment:
Thanks for adding the `validateCreateDeviceParams` validator! It looks like it's not getting called here yet, though - please wire it up!